tests/1_engine/rex_engine_unit_test/src/test_path.cpp

```diff
@@ -0,0 +1,525 @@
+  // No proper way atm to create or delete files, unless using the vfs
+  rsl::cout << "COULDN'T TEST FILE CREATION TIME\n";
+
+  // wait 1 tail
+  using namespace rsl::chrono_literals;
+  rsl::this_thread::sleep_for(1s);
+  
+  // query the creation time again, it should still be the same
+  creation_time = rex::path::get_creation_time(random_filename);
+}
+
+TEST_CASE("File Modification Time")
+{
+  // get the current time
+  auto now = rsl::chrono::high_resolution_clock::now().time_since_epoch();
+
+  // create a file that doesn't exist yet
+  rsl::string random_filename = rex::path::random_filename();
+
+  // query the modification time and compare it to the original time with a threshold
+  auto creation_time = rex::path::get_modification_time(random_filename);
+  // No proper way atm to create or delete files, unless using the vfs
+  rsl::cout << "COULDN'T TEST FILE MODIFICATION TIME\n";
+
+  // wait 1 tail
+  using namespace rsl::chrono_literals;
+  rsl::this_thread::sleep_for(1s);
+
+  // query the modification time again, it should still be the same
+  creation_time = rex::path::get_modification_time(random_filename);
+
+  // get the current time again
+  now = rsl::chrono::high_resolution_clock::now().time_since_epoch();
+
+  // Modify the file
+  // No proper way atm to modify a file, unless using the vfs
+
+  // query the modification time again, it should now be different
+  creation_time = rex::path::get_modification_time(random_filename);
+}
+
+TEST_CASE("File Access Time")
+{
+  // get the current time
+  auto now = rsl::chrono::high_resolution_clock::now().time_since_epoch();
+
+  // create a file that doesn't exist yet
+  rsl::string random_filename = rex::path::random_filename();
+
+  // query the modification time and compare it to the original time with a threshold
+  auto creation_time = rex::path::get_modification_time(random_filename);
+  // No proper way atm to create or delete files, unless using the vfs
+  rsl::cout << "COULDN'T TEST FILE ACCESS TIME\n";
+
+  // wait 1 tail
+  using namespace rsl::chrono_literals;
+  rsl::this_thread::sleep_for(1s);
+
+  // query the access  time again, it should still be the same
+  creation_time = rex::path::get_access_time(random_filename);
+
+  // get the current time again
+  now = rsl::chrono::high_resolution_clock::now().time_since_epoch();
+
+  // Modify the file
+  // No proper way atm to modify a file, unless using the vfs
+
+  // query the access  time again, it should now be different
+  creation_time = rex::path::get_access_time(random_filename);
+
+  // get the current time again
+  now = rsl::chrono::high_resolution_clock::now().time_since_epoch();
+
+  // Read the file
+  // No proper way atm to read a file, unless using the vfs
+
+  // query the access time again, it should now be different
+  creation_time = rex::path::get_access_time(random_filename);
+}
+
+TEST_CASE("File Size")
+{
+  CHECK(rex::path::get_file_size("file_0_bytes.txt") == 0);
```

Comment:
Not really sure if this should be part of "path" path should manipulate paths not check file_size this seems more like something the VFS or some IO system should know about.